dts/msm8974/msm8974-samsung-r10.dts

```diff
@@ -9,8 +9,8 @@
 	qcom,msm-id = <0xC208FF01 10 0x10000>;
 
 	kltechn {
-		model = "Samsung Galaxy S5 China Unicom (Duos) (SM-G9006V/W)";
+		model = "Samsung Galaxy S5 China Unicom/Mobile (Duos) (SM-G9006V/W,SM-G9008V/W)";
 		compatible = "samsung,klte", "qcom,msm8974", "lk2nd,device";
-		lk2nd,match-bootloader = "G9006*";
+		lk2nd,match-bootloader = "G9006*", "G9008*";
```

Comment:
So far in lk2nd we were handling device variants as separate devices. There are aleady multiple examples where we have dedicated entries for xx006, xx008 and xx009 variants of devices for different Chinese carriers.

I'm afraid that making the "model" line in lk2nd longer would be confusing and it already seems there are three devices in this lineup so with this setup it would grow more. On top of that, sometimes those variants may have slight hardware differences (I believe you've already found one between G9008 and already existing G900F) so I'd like to keep them separate in light of future lk2nd where we would provision a correct DTB based on this data.

I suggest adding those three models separately, adding some suffix to distinguish them, i.e. like done here:

https://github.com/msm8916-mainline/lk2nd/blob/373284a5718d2aa2e01b80f9611228a21e93d278/dts/msm8916/msm8916-samsung-r03.dts#L24-L43

I believe so far we never needed to care about sim slot count since it's managed by modem and not Linux, so keeping a model for both single sim and Duos like this one was is fine though.

Reply:
Interesting, as the original author seems to be not free until weekend, should I try to do this?

Reply:
Is it needed to have different compatible string for different carrier variants?

Reply:
I'd prefer to see something like this to denote the hardware is "almost compatible" with the base klte:
```
 	compatible = "samsung,klte-chnctc", "samsung,klte", "qcom,msm8974", "lk2nd,device"; 
```
and I'd appreciate if you can find what Samsung calls them in the downstream files

Reply:
BTW I assume there should be kltechn and klte (because the LED/GPIO expander I2C bus is located differently on kltechn). In addition, the downstream DT seems to be the same for all China LTE variants (China Telecom has a 3G variant, but this is another story and won't be supported here).

Reply:
https://github.com/LineageOS/android_kernel_samsung_msm8974/blob/lineage-18.1/arch/arm/boot/dts/msm8974pro/msm8974pro-ac-sec-kchn-r10.dts The downstream just calls it kchn, without any carrier name (there's another kctc, but kctc is for the 3G version of China Telecom, G9009D.)

Reply:
Interesting, then I'd be inclined to say it's
```
 	compatible = "samsung,kchn-ctc", "samsung,kchn", "samsung,klte", "qcom,msm8974", "lk2nd,device"; 
```

meaning: The exact device is "China Telecom S5" that is a derivative of "China region S5" that is a derivative of "S5".

Then for upstream Linux I'd expect to see
```
 	compatible = "samsung,kchn", "samsung,klte", "qcom,msm8974";
```
for the dt that fixes up the gpio expander, given you say there should be no more difference.

Reply:
or ugh, maybe I don't understand exactly what the naming is there but I hope you got my point: in lk2nd we can add dedicated devices for each "too specific" model, but in linux we want to group them by real hardware differences. I will leave the exact compatibles and names at your discretion

Reply:
Then how should China Mobile/Unicom get named, and the "lte" string should appear before "chn" (as the downstream klte device tree is just labelled as "k", thus "kchn" should be kltechn, which is also the named used by LineageOS)

BTW China Mobile is usually named "cmcc", but there's not a properly abbr for China Unicom.

Reply:
In addition, if trying to add for "too specific" models, maybe the V and W variants should be splitted too (well, it visually, represents how many SIM slots are available, V for solo and W for duo)

Reply:
> maybe the V and W variants should be splitted too

As I've mentioned earlier, we usually don't differentiate single/duos devices - they are usually only different in a flag in modem firmware + a bit of different plastic and extra three passive components placed on the pcb, with the device being completely the same otherwise.

You could argue it's a somewhat arbitrary distinction but this is what we historically did in lk2nd and Linux so far...